server/master/src/main/java/org/apache/accumulo/master/TimeoutTaskExecutor.java

```diff
@@ -152,15 +172,24 @@ public void close() {
   }
 
   /*
-   * A wrapper for keeping a callable and its future together.
+   * A wrapper for a Callable that keeps additional information. This tracks if the callable has been started, has completed (either finished or cancelled), and
+   * keeps the associated future.
    */
-  private class WrappedTask {
+  private class WrappedTask implements Callable<T> {
     public final C callable;
-    public final Future<T> future;
 
-    public WrappedTask(C callable, Future<T> future) {
+    public volatile boolean hasStarted = false;
+    public boolean hasCompleted = false;
+    public Future<T> future;
+
+    public WrappedTask(C callable) {
       this.callable = callable;
-      this.future = future;
+    }
+
+    @Override
+    public T call() throws Exception {
+      hasStarted = true;
```

Comment:
What do you think about recording the time here instead of having a boolean?  Then that could be used to determine how long to wait for the task.

Reply:
Sounds good!